### PR TITLE
ref(compactSelect): Small fixes & UI updates

### PR DESCRIPTION
**Pull request by vuluongj20**
Making a few bug fixes & UI updates. See comments below for more details.

### CODE REVIEW

File: static/app/components/dropdownButtonV2.tsx [hunk 1]
```diff
@@ -47,11 +47,7 @@ const DropdownButton = forwardRef<
       {prefix && <LabelText>{prefix}</LabelText>}
       {children}
       {showChevron && (
-        <StyledChevron
-          size="10px"
-          direction={isOpen ? 'up' : 'down'}
-          aria-hidden="true"
-        />
+        <StyledChevron size="xs" direction={isOpen ? 'up' : 'down'} aria-hidden="true" />
```
Comment (vuluongj20) [inline static/app/components/dropdownButtonV2.tsx:50]: Using a larger iconChevron (`12px`), which is more in line with our other trigger buttons.

File: static/app/components/forms/selectOption.tsx [hunk 2]
```diff
@@ -12,47 +13,52 @@ function SelectOption(props: Props) {
   const {label, data, selectProps, isMulti, isSelected, isFocused, isDisabled} = props;
   const {showDividers, verticallyCenterCheckWrap} = selectProps;
   const {
+    value,
     details,
+    tooltip,
+    tooltipDelay,
```
Comment (evanpurkhiser) [inline static/app/components/forms/selectOption.tsx:19]: I would have `tooltipOptions` instead probably

File: static/app/components/forms/compactSelect.tsx [hunk 3]
```diff
@@ -178,41 +178,51 @@ function CompactSelect<OptionType extends GeneralSelectValue = GeneralSelectValu
...
-        <ButtonLabel>{firstOptionLabel}</ButtonLabel>
-        {optionSet.length > 1 && <StyledBadge text={`+${optionSet.length - 1}`} />}
+        <ButtonLabel>{selectedOptions[0]?.label ?? ''}</ButtonLabel>
+        {selectedOptions.length > 1 && (
+          <StyledBadge text={`+${selectedOptions.length - 1}`} />
+        )}
       </Fragment>
     );
-  }
+  }, [options, valueProp, internalValue]);
```
Comment (vuluongj20) [inline static/app/components/forms/compactSelect.tsx:198]: Wrapping `getLabel()` inside a `useCallback()`, so it doesn't get re-defined on every render.

File: static/app/components/forms/compactSelect.tsx [hunk 4]
```diff
@@ -178,41 +178,51 @@ function CompactSelect<OptionType extends GeneralSelectValue = GeneralSelectValu
...
+  // If ResizeObserver is not available, manually update the width
+  // when any of [trigger, triggerLabel, triggerProps] changes.
   useEffect(() => {
-    // Wait until the trigger label has been updated before calculating the
-    // new width
-    setTimeout(() => {
-      const newTriggerWidth = triggerRef.current?.offsetWidth;
-      newTriggerWidth ?? setTriggerWidth(newTriggerWidth);
-    }, 0);
-  }, [triggerRef.current, internalValue]);
```
Comment (vuluongj20) [inline static/app/components/forms/compactSelect.tsx:215]: There was a memory leak with the `setTimeout`. If the component gets unmounted after the deps array changes but before the contents of `setTimeout` executes, then we would have `setTriggerWidth` called inside an unmounted component. 
An easy way to fix this is to provide a cleanup function - `return () => clearTimeout(timeoutId)`. But I figured just using `await new Promise(resolve => window.setTimeout(resolve));` and not the timeout was a better solution: less indentation, no memory leak.

File: static/app/components/forms/selectOption.tsx (hunk 2 above)
Comment (vuluongj20) [reply static/app/components/forms/selectOption.tsx]: Good point https://github.com/getsentry/sentry/pull/33983/commits/08eeec89756e959c6dba378f7392b86f9549f802

File: static/app/components/forms/selectOption.tsx [hunk 5]
```diff
@@ -12,47 +13,52 @@ function SelectOption(props: Props) {
...
-      <InnerWrap isFocused={isFocused} isDisabled={isDisabled}>
-        <Indent isMulti={isMulti} centerCheckWrap={verticallyCenterCheckWrap}>
-          <CheckWrap isMulti={isMulti} isSelected={isSelected}>
-            {isSelected && (
-              <IconCheckmark
-                size={isMulti ? 'xs' : 'sm'}
-                color={isMulti ? 'white' : undefined}
-              />
+    <selectComponents.Option className="select-option" {...props}>
+      <Tooltip containerDisplayMode="block" title={tooltip} {...tooltipOptions}>
```
Comment (evanpurkhiser) [inline static/app/components/forms/selectOption.tsx:28]: Can you use the `skipWrapper` prop here?

**Review by evanpurkhiser — APPROVED** [review]
Looking pretty good, left a small comment

File: static/app/components/forms/selectOption.tsx (hunk 5 above)
Comment (vuluongj20) [reply static/app/components/forms/selectOption.tsx]: Oh nice didn't know that was available! - https://github.com/getsentry/sentry/pull/33983/commits/4d1447f6ac5107ed10dbb6f59f81fbc7d281ff9a

**Pull request merged**